src/unit.c

```diff
@@ -218,76 +114,60 @@ static double evaluateNullUnit(double value, double thisCM,
  */
 double pureNullUnitValue(SEXP unit, int index)
 {
-    double result = 0;
-    if (isUnitArithmetic(unit)) {
-	int i;
-	if (addOp(unit)) {
-	    result = pureNullUnitValue(arg1(unit), index) + 
-		pureNullUnitValue(arg2(unit), index);
-	}
-	else if (minusOp(unit)) {
-	    result = pureNullUnitValue(arg1(unit), index) - 
-		pureNullUnitValue(arg2(unit), index);
-	}
-	else if (timesOp(unit)) {
-	    result = REAL(arg1(unit))[index] * 
-		pureNullUnitValue(arg2(unit), index);
-	}
-	else if (minFunc(unit)) {
-	    int n = unitLength(arg1(unit));
-	    double temp = DBL_MAX;
-	    result = pureNullUnitValue(arg1(unit), 0);
-	    for (i=1; i<n; i++) {
-		temp = pureNullUnitValue(arg1(unit), i);
-		if (temp < result)
-		    result = temp;
-	    }
-	} 
-	else if (maxFunc(unit)) {
-	    int n = unitLength(arg1(unit));
-	    double temp = DBL_MIN;
-	    result = pureNullUnitValue(arg1(unit), 0);
-	    for (i=1; i<n; i++) {
-		temp = pureNullUnitValue(arg1(unit), i);
-		if (temp > result)
-		    result = temp;
-	    }
-	} 
-	else if (sumFunc(unit)) {
-	    int n = unitLength(arg1(unit));
-	    result = 0.0;
-	    for (i=0; i<n; i++) {
-		result += pureNullUnitValue(arg1(unit), i);
-	    }
-	}
-	else 
-	    error(_("unimplemented unit function"));
-    } else if (isUnitList(unit)) {
-	/*
-	 * Recycle if necessary;  it is up to the calling code
-	 * to limit indices to unit length if desired
-	 */
-	int n = unitLength(unit);
-	result = pureNullUnitValue(VECTOR_ELT(unit, index % n), 0);
-    } else
-	result = unitValue(unit, index);
-    return result;
+  double result = 0;
```

Comment:
Does the comment above also need an update? (Maybe worth grepping for `unitList`/`unitArithmetic`?)